### PR TITLE
Introduce basic Detox Instruments demo app 

### PR DESCRIPTION
**Pull request by rotemmiz**
- [ ] This is a small change 
- [ ] This change has been discussed in issue #<?> and the solution has been agreed upon with maintainers.

---

**Description:**
Introduce a demo app that includes an integration of detox-instruments for demo and testing purposes inside detox repo.

### CODE REVIEW

File: examples/demo-react-native-detox-instruments/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "test:ios": "detox test --configuration ios.sim.debug --record-performance all",
+    "e2e:ios": "npm run build:ios && npm run test:ios"
+  },
+  "dependencies": {
+    "react": "16.8.4",
+    "react-native": "0.59.x"
+  },
+  "devDependencies": {
+    "detox": "^12.4.1",
+    "detox-instruments": "1.x.x",
```
Comment (LeoNatan) [inline examples/demo-react-native-detox-instruments/package.json]: This is unsupported. Please do not include it on any external project.

Comment (LeoNatan) [reply examples/demo-react-native-detox-instruments/package.json]: The only supported method is `brew cask`.

Comment (rotemmiz) [reply examples/demo-react-native-detox-instruments/package.json]: I can remove it from the examples folder, but this as a test project

Comment (LeoNatan) [reply examples/demo-react-native-detox-instruments/package.json]: It doesn’t matter. Detox Instruments should not be installed from npm. It should be installed as described in the developer guide.

**Comment by LeoNatan** [comment]
As discussed, I will make a change to install Detox Instruments from `brew cask`. In order to test custom path, the script will provide the default path as a custom path.

**Pull request merged**